File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
      let maxp = 0;
      let mini = prices[0];
        
        for (let i = 1; i < prices.length; i++) {  
            let c = prices[i] - mini;
            maxp = Math.max(maxp, c);
            mini = Math.min(mini, prices[i]);
        }
        
        return maxp;
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
      int maxp = 0;
        int mini = prices[0];
        
        for (int i = 1; i < prices.length; i++) {  
            int c = prices[i] - mini;
            maxp = Math.max(maxp, c);
            mini = Math.min(mini, prices[i]);
        }
        
        return maxp;
    }
}